pkg/e2e/infra/infra.go

```diff
@@ -197,232 +229,88 @@ var _ = g.Describe("[Feature:Machines] Managed cluster should", func() {
-		err = deleteMachine(client, workerMachine)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By(fmt.Sprintf("waiting for node object %q to go away", workerNode.Name))
-		nodeList := corev1.NodeList{}
-		o.Eventually(func() bool {
-			if err := client.List(context.TODO(), &nodeList); err != nil {
-				glog.Errorf("Error querying api for nodeList object: %v, retrying...", err)
-				return false
-			}
-			for _, n := range nodeList.Items {
-				if n.Name == workerNode.Name {
-					glog.Infof("Node %q still exists. Node conditions are: %v", workerNode.Name, workerNode.Status.Conditions)
-					return false
-				}
-			}
-			return true
-		}, e2e.WaitLong, 5*time.Second).Should(o.BeTrue())
+		e2e.WaitForMachineDelete(client, machine)
 
-		g.By(fmt.Sprintf("waiting for new node object to come up"))
-		err = waitForClusterSizeToBeHealthy(client, initialClusterSize)
-		o.Expect(err).NotTo(o.HaveOccurred())
+		e2e.WaitForMachineSet(client, machineSet.GetName())
 	})
 
 	g.It("grow and decrease when scaling different machineSets simultaneously", func() {
-		client, err := e2e.LoadClient()
-		o.Expect(err).NotTo(o.HaveOccurred())
 		scaleOut := 2
 
-		// Anything we create we must cleanup
-		var cleanupObjects []runtime.Object
-		defer func() {
-			for _, obj := range cleanupObjects {
-				if err := deleteObject(client, obj); err != nil {
-					glog.Errorf("[cleanup] error deleting object: %v", err)
-				}
-			}
-		}()
-
-		g.By("checking existing cluster size")
-		existingClusterSize, err := getClusterSize(client)
+		g.By("Scale down pre-created MachineSet")
+		err := scaleMachineSet(machineSet.Name, 0)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		g.By("getting worker machineSets")
-		existingMachineSets, err := e2e.GetMachineSets(client)
-		o.Expect(err).NotTo(o.HaveOccurred())
-		o.Expect(len(existingMachineSets)).To(o.BeNumerically(">=", 1))
-
-		// Create transient machinesets with replicas==0
-		machineSets := make([]*mapiv1beta1.MachineSet, scaleOut)
-		randomUUID := string(uuid.NewUUID())
-		for i := 0; i < scaleOut; i++ {
-			targetMachineSet := existingMachineSets[i%len(existingMachineSets)]
-			machineSetName := fmt.Sprintf("e2e-%s-w-%d", randomUUID[:5], i)
-			glog.Infof("Creating transient MachineSet %q", machineSetName)
-			machineSets[i] = e2e.NewMachineSet(
-				targetMachineSet.Labels[e2e.ClusterKey],
-				targetMachineSet.Namespace,
-				machineSetName,
-				targetMachineSet.Spec.Selector.MatchLabels,
-				targetMachineSet.Spec.Template.ObjectMeta.Labels,
-				&targetMachineSet.Spec.Template.Spec.ProviderSpec,
-				0) // zero replicas
-			machineSets[i].Spec.Template.Spec.ObjectMeta.Labels = map[string]string{
-				e2e.WorkerNodeRoleLabel: "",
-			}
-			o.Expect(client.Create(context.TODO(), machineSets[i])).Should(o.Succeed())
-			cleanupObjects = append(cleanupObjects, runtime.Object(machineSets[i]))
-		}
+		g.By("Creating a second MachineSet")
+		machineSetParams := buildMachineSetParams(client, 0)
```

Comment:
I think this whole test can be simplified by something like:
```
machineSet2, err := e2e.CreateMachineSet(client, machineSetParams)
e2e.WaitForMachineSet(client, machineSet2.GetName())
scaleMachineSet(machineSet1, 0)
scaleMachineSet(machineSet2, 3)
e2e.WaitForMachineSet(client, machineSet1.GetName())
e2e.WaitForMachineSet(client, machineSet2.GetName())
```
Once we run the tests in parallel we can probably just drop it